microprot/scripts/pdb_search.py

```diff
@@ -0,0 +1,393 @@
+            else:
+                block[name]['sequence'] += lines[idx][startCol:endCol]
+            if lines[idx][endCol:] != '':
+                fields = lines[idx][endCol:].split()
+                block[name]['end'] = int(fields[0])
+                block[name]['totallen'] = int(fields[1][1:-1])
+            idx += 1
+        idx += 2
+
+    hit['alignment'] = block
+
+    return hit
+
+
+def parse_pdb_match(filename):
+    """ Parse an HHsearch output file.
+
+    Parameters
+    ----------
+    filename : str
+        Path to the HHsearch output file that should be parsed.
+
+    Returns
+    -------
+    A list of HHsearch hits. Each hit is a dict, holding all its information.
+
+    Raises
+    ------
+    IOError
+        If the file cannot be read.
+    """
+    hits = []
+    try:
+        fh = open(filename, 'r')
+        line = ""
+
+        # read until header of summary table is found
+        while(len(set(line.rstrip().split()) & set(_HEADER)) < 8):
+            line = fh.readline()
+        line = fh.readline()  # summary table header line
+        # read all summary lines
+        while(line != '\n'):
+            hits.append(_parse_hit_summary_line(line))
+            line = fh.readline()
+
+        # read the alignments
+        reachedEOF = False
+        alignments = []
+        block = ''
+        while(not reachedEOF):
+            line = fh.readline()
+            reachedEOF = line == ''
+            if line.startswith('No ') & (len(line.split()) == 2):
+                if block != '':
+                    alignments.append(_parse_hit_block(block))
+                block = line
+            else:
+                block += line
+        if block != '':
+            alignments.append(_parse_hit_block(block))
+
+        # merge hit summary and according alignment
+        for idx in range(len(hits)):
+            for key in alignments[idx]:
+                hits[idx][key] = alignments[idx][key]
+            # duplicate information, but more precicse on other place
+            del hits[idx][_HEADER[2]]  # Prob
+            del hits[idx][_HEADER[8]]  # Query HMM
+            del hits[idx][_HEADER[9]]  # Template HMM
+            del hits[idx][_HEADER[10]]  # match states
+
+        fh.close()
+        return hits
+    except IOError:
+        raise IOError('Cannot read file "%s"' % filename)
+
+
+def select_hits(hits, e_value_threshold=0.001):
+    """ Picking HHsearch hits from the list of all hits.
+
+    We take those hits that a) have an e-Value <= e_value_threshold and b)
+    does not overlap with any previously picked hit.
+
+    Parameters
+    ----------
+    hits: [dict]
+        The sorted list of hits from an HHsearch.
+    e_value_threshold: float
+        The threshold for a maximal e-Value for a hit to pick. Default: 0.001
+
+    Returns
+    -------
+    [dict] a potentially smaller list of HHsearch hits.
+    """
+    good_hits = []
+
+    for hit in hits:
```

Comment:
Unlikely, because longer hits would get a lower E-value than similar shorter hits, i.e. 2-element subset of a longer sequence would have a lower E-value. If that's what you have in mind.